### ISSUE TITLE
bug: appium-fake-driver

### ISSUE THREAD

**Issue by timwright12** [appium/appium#18746]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I'm getting a critical vulnerability error with xmldom, which is used in `appium-fake-driver`: https://www.npmjs.com/advisories/1091573 

It looks like xmldom was deprecated in favor of @xmldom/xmldom (because the package was deprecated and moved we can't use yarn resolutions to clear it out). It looks like you're working on a 2.0 release of appium, which is great, but I would imagine since it's a full version bump that there will be breaking chnages and anyone on 1.22.3 (current npm version) may need to do an amount of testing before updating and will have to living with the critical xmldom vulnerability in the meantime.

It would be great if a quick patch for appium-fake-driver could be pushed out in the meantime to help teams in that situation.

### Expected Behavior

No critical vulnerability errors.

### Minimal Reproducible Example

yarn audit or npm audit on v1.22.3 will pull up the issue

### Environment

Just running yarn on a mac Node version 16

**Comment by eglitise** [comment]
Appium 1.x is no longer supported since 2022, so my suggestion would be to update to Appium 2. The current version of `appium-fake-driver` (or rather, `@appium/fake-driver`) is already using the latest version of `@xmldom/xmldom` and is therefore safe to update to.
However, as you said, Appium 2 comes with breaking changes. [There is a migration guide to help you with this upgrade](https://appium.io/docs/en/latest/guides/migrating-1-to-2/).

**Comment by jlipps** [comment]
@timwright12 additionally, why are you using appium-fake-driver? it's really just for appium's own internal testing. it should (hopefully) not be included in any other appium packages.

**Comment by timwright12** [comment]
@jlipps It may not be actually used, but because it's included in a sub-dependancy, it shows up in in the security audit, so if you want a clear audit report you can't get it. For me, I'm trying to turn my organization's repo from private to public and one of the stipulations there is that all security vulnerabilities are cleared before we can turn open source.

**Comment by mykola-mokhnach** [comment]
@jlipps AFAIK it was automatically shipped as a part of Appium1 package deps, which is not happening anymore in Appium 2 world, where drivers are selectable: https://github.com/appium/appium/blob/465565d3fdb0f7e0562b4eb4806f28aa994b8369/package.json#L47

**Comment by jlipps** [comment]
@timwright12 if you're talking about Appium 1, we aren't shipping any more versions of Appium 1, so you should just upgrade to Appium 2. But let me know where it's a subdependency of something else if it still is. Fake driver shouldn't be a prod dep of any useful package (though it might be a dev dep used for testing).

**Comment by timwright12** [comment]
from my description:

> It looks like you're working on a 2.0 release of appium, which is great, but I would imagine since it's a full version bump that there will be breaking chnages and anyone on 1.22.3 (current npm version) may need to do an amount of testing before updating and will have to living with the critical xmldom vulnerability in the meantime

**Comment by jlipps** [comment]
the only reason that 1.22 has been the current version is we've been working on the docs for a year. appium 2 from a technical perspective is ready to go, and has been for some time. i recommend migrating now. there are basically no driver-specific breaking changes associated with appium 2.

**Comment by jlipps** [comment]
closing this as related to appium 1 which is no longer supported

**Comment by jlipps** [comment]
Correct. That is the most accurate guide that exists. Let us know if we missed anything there.
